.github/workflows/PR-build.yml

```diff
@@ -134,9 +141,89 @@ jobs:
         run: |
           ARCH=amd64 TARGET_SUPPORTED_ARCH=x86_64 DEST=build/packages/debian/amd64 tools/packaging/debian/create_deb.sh
           ARCH=arm64 TARGET_SUPPORTED_ARCH=aarch64 DEST=build/packages/debian/arm64 tools/packaging/debian/create_deb.sh
-   
+
+  packaging-image:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Cache Docker Image
+        id: cached_image
+        uses: actions/cache@v2
+        with:
+          key: "cached_image_${{ github.run_id }}"
+          path: "${{ env.PACKAGING_ROOT }}"
+
+      - name: Build Docker Image
```

Comment:
Use the official action for building

https://github.com/docker/build-push-action/blob/master/docs/advanced/cache.md

It has docs on how to use cache natively too instead of a separate save step